js/app.js

```diff
@@ -412,6 +422,15 @@ App.prototype.runAnalyzer = function() {
 
 	this.analyzerData.keyword = keyword;
 
+	// The new researcher
+	if ( isUndefined( this.researcher ) ) {
+		this.researcher = new Researcher( this.paper );
+		this.pluggable._addPluginTests( this.researcher );
+	} else {
+		this.researcher.updatePaper( this.paper );
+		this.pluggable._addPluginTests( this.researcher );
```

Comment:
This won't work on the researcher, the function `_addPluginTests` tries to call a `addAnalysis` function on the object passed as an argument, but that function doesn't exist on the Researcher.